cli/src/cli_util.rs

```diff
@@ -1433,7 +1433,21 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
                 print_failed_git_export(ui, &failed_branches)?;
             }
 
-            self.user_repo = ReadonlyUserRepo::new(tx.commit("snapshot working copy"));
+            const DESCRIPTION: &str = "snapshot working copy";
+            const TAG: &str = "snapshots";
+
+            if self.settings.squash_consecutive_snapshots() {
+                if let [parent_op] = tx.parent_ops() {
+                    let meta = &parent_op.store_operation().metadata;
+                    if meta.description == DESCRIPTION {
+                        let tag = meta.tags.get(TAG).and_then(|s| s.parse().ok()).unwrap_or(1);
+                        tx.replace_parent()?;
+                        tx.set_tag(TAG.into(), (tag + 1).to_string())
```

Comment:
It's probably better to add `snapshots=1` (or `is_snapshot=1`) tag to the first snapshot operation, and check it instead of comparing the description.

Reply:
Yes, just checking for the presence of "snapshots" tag is immediately way better, didn't think of that (since I only later realised ops have arbitrary tags, which I could use to keep track of the number)

I'll do that when I get back to my pc 👍

Reply:
Yes, tagging snapshots seems useful in general. We've talked about doing that before but I don't remember in what context. Maybe it could be useful for showing snapshot operations differently in the operation log. Could also be useful for opsets.

Reply:
Just setting "snapshots" to 1 by default, and checking for it's presence, I think that should be ok